src/blueapi/config.py

```diff
@@ -89,83 +90,16 @@ class ApplicationConfig(BlueapiBaseModel):
 
-    def __eq__(self, other: object) -> bool:
-        if isinstance(other, ApplicationConfig):
-            return (
-                (self.stomp == other.stomp)
-                & (self.env == other.env)
-                & (self.logging == other.logging)
-                & (self.api == other.api)
-            )
-        return False
-
-
-C = TypeVar("C", bound=BaseModel)
-
-
-class ConfigLoader(Generic[C]):
-    """
-    Small utility class for loading config from various sources.
-    You must define a config schema as a dataclass (or series of
-    nested dataclasses) that can then be loaded from some combination
-    of default values, dictionaries, YAML/JSON files etc.
-    """
-
-    def __init__(self, schema: type[C]) -> None:
-        self._adapter = TypeAdapter(schema)
-        self._values: dict[str, Any] = {}
-
-    def use_values(self, values: Mapping[str, Any]) -> None:
-        """
-        Use all values provided in the config, override any defaults
-        and values set by previous calls into this class.
-
-        Args:
-            values (Mapping[str, Any]): Dictionary of override values,
-                                        does not need to be exhaustive
-                                        if defaults provided.
-        """
-
-        def recursively_update_map(old: dict[str, Any], new: Mapping[str, Any]) -> None:
-            for key in new:
-                if (
-                    key in old
-                    and isinstance(old[key], dict)
-                    and isinstance(new[key], dict)
-                ):
-                    recursively_update_map(old[key], new[key])
-                else:
-                    old[key] = new[key]
-
-        recursively_update_map(self._values, values)
-
-    def use_values_from_yaml(self, path: Path) -> None:
-        """
-        Use all values provided in a YAML/JSON file in the
-        config, override any defaults and values set by
-        previous calls into this class.
-
-        Args:
-            path (Path): Path to YAML/JSON file
-        """
-
-        with path.open("r") as stream:
-            values = yaml.load(stream, yaml.Loader)
-        self.use_values(values)
-
-    def load(self) -> C:
-        """
-        Finalize and load the config as an instance of the `schema`
-        dataclass.
-
-        Returns:
-            C: Dataclass instance holding config
-        """
-
-        try:
-            return self._adapter.validate_python(self._values)
-        except ValidationError as exc:
-            error_details = "\n".join(str(e) for e in exc.errors())
-            raise InvalidConfigError(
-                f"Something is wrong with the configuration file: \n {error_details}"
-            ) from exc
+    model_config = SettingsConfigDict(
+        env_nested_delimiter="__", yaml_file=DEFAULT_PATH, yaml_file_encoding="utf-8"
+    )
+
+    @classmethod
+    def customize_sources(cls, init_settings, env_settings, file_secret_settings):
```

Comment:
Looks like `settings_customise_sources` [in the docs](https://docs.pydantic.dev/latest/concepts/pydantic_settings/#the-basics)?

Reply:
```python
class Settings(BaseSettings):
    foobar: str
    nested: Nested
    model_config = SettingsConfigDict(toml_file='config.toml')

    @classmethod
    def settings_customise_sources(
        cls,
        settings_cls: Type[BaseSettings],
        init_settings: PydanticBaseSettingsSource,
        env_settings: PydanticBaseSettingsSource,
        dotenv_settings: PydanticBaseSettingsSource,
        file_secret_settings: PydanticBaseSettingsSource,
    ) -> Tuple[PydanticBaseSettingsSource, ...]:
        return (TomlConfigSettingsSource(settings_cls),)
```

Looking at how a toml file is used by an equivalent 

And 

> settings_customise_sources takes four callables as arguments and returns any number of callables as a tuple. In turn these callables are called to build the inputs to the fields of the settings class.

I think you need the signature to exactly match what is shown?

Reply:
yeah, it needs to be an override. Not sure why did this got changes